plugins/search-backend-module-pg/api-report.md

```diff
@@ -95,8 +101,13 @@ export class PgSearchEngine implements SearchEngine {
   ): void;
   // (undocumented)
   static supported(database: PluginDatabaseManager): Promise<boolean>;
+  // Warning: (ae-forgotten-export) The symbol "PgSearchHighlightConfig" needs to be exported by the entry point index.d.ts
```

Comment:
Do we need both XConfig and XOptions as above? I presume one is the raw config representation, but maybe we can forego that one entirely (at least in the public interface)? If we move the logic for what defaults to apply, the Config type may not be needed. May make sense to make the constructor private in either case - I know it's useful to have visible for tests, but you could essentially do `new (PgSearchEngine  as any)(...)` or whatnot there if you don't mind the ugliness.

Reply:
I settled on keeping the options object and removing the config object